### PR TITLE
chore: Use shared devDependencies in schematics

### LINKED ISSUE #10429: Avoid dependencies in @spartacus/schematics
We dropped @angular/pwa as a dependency in schematics in #10386

We should do it also for localize and express-engine library and install them in created application before executing their schematics.

Also fix comments:
https://github.com/SAP/spartacus/pull/10426#discussion_r543472241
https://github.com/SAP/spartacus/pull/10426#discussion_r543473733

### PR DESCRIPTION
**Pull request by marlass**
We should keep all dependencies used by our libraries in top level `node_modules`. External schematics should be added to schematics as `dependencies`. Schematics should not have any `devDependencies`. All used tools should be listed in root `package.json` file. 

PWA schematics should be similar to the state before #10426

Closes #10429

### CODE REVIEW

File: feature-libs/organization/package.json [hunk 1]
```diff
@@ -22,25 +21,15 @@
...
   "peerDependencies": {
+    "@angular-devkit/schematics": "^10.1.0",
     "@angular/common": "^10.1.0",
     "@angular/core": "^10.1.0",
-    "@angular-devkit/schematics": "^10.1.0",
     "@angular/router": "^10.1.0",
-    "bootstrap": "^4.0",
-    "rxjs": "^6.6.0",
     "@spartacus/core": "3.0.0",
+    "@spartacus/schematics": "3.0.0",
```
Comment (znikola) [inline feature-libs/organization/package.json:29]: By doing this, is it possible to read "live" changes from the projects/schematics during development?

File: projects/schematics/src/add-pwa/index.ts [hunk 2]
```diff
@@ -90,34 +84,6 @@ function updateAppModule(options: any): Rule {
...
-  return (tree: Tree, context: SchematicContext) => {
-    const dependencies = [
-      {
-        type: NodeDependencyType.Dev,
-        version: '^0.1001.0',
-        name: '@angular/pwa',
-      },
-    ];
-    return chain([
-      addPackageJsonDependencies(dependencies),
```
Comment (znikola) [inline projects/schematics/src/add-pwa/index.ts:102]: Why was this removed?
Shouldn't we install the PWA dep in the customer's project?

File: feature-libs/organization/package.json (hunk 1 above)
Comment (marlass) [reply feature-libs/organization/package.json:29]: I didn't changed anything here 👀 
I'll check, but it's just prettier reorder.

File: projects/schematics/src/add-pwa/index.ts (hunk 2 above)
Comment (marlass) [reply projects/schematics/src/add-pwa/index.ts:102]: Installation happens when you install @spartacus/schematics as @angular/pwa is now specified as `dependency`.
And then we only want to run schematics which have it specified to not add it in dependencies (https://github.com/angular/angular-cli/blob/master/packages/angular/pwa/package.json).

I guess that's the goal: run `ng add @angular/pwa`, but not add it to project deps.

I checked and saw that the changes from @angular/pwa ar applied, but still would like a second confirmation.

Main idea behind external schematics run in our schematics is to specify them as dependencies, but not explicitly install them in customer application, but rely on mechanism that npm install all required deps by a package and let the external schematics decide if it should be added to package.json (localize, express-ngine) or not (pwa).

**Review by znikola — APPROVED** [review]
